File: 1603-design-parking-system/1603-design-parking-system.c
typedef struct {
    int bigSize;
    int mediumSize;
    int smallSize;
} ParkingSystem;

ParkingSystem* parkingSystemCreate(int big, int medium, int small) {
    ParkingSystem *obj = (ParkingSystem*)malloc(sizeof(ParkingSystem));
    obj->bigSize = big;
    obj->mediumSize = medium;
    obj->smallSize = small;
    return obj;
}

bool parkingSystemAddCar(ParkingSystem* obj, int carType) {
    if(carType == 1){
        if(!obj->bigSize)
            return false;
        obj->bigSize--;
        return true;
    }
    else if(carType == 2){
        if(!obj->mediumSize)
            return false;
        obj->mediumSize--;
        return true;
    }
    //carType == 3
    if(!obj->smallSize)
        return false;
    obj->smallSize--;
    return true;
}

void parkingSystemFree(ParkingSystem* obj) {
    free(obj);
}

/**
 * Your ParkingSystem struct will be instantiated and called as such:
 * ParkingSystem* obj = parkingSystemCreate(big, medium, small);
 * bool param_1 = parkingSystemAddCar(obj, carType);
 
 * parkingSystemFree(obj);
*/

File: 1603-design-parking-system/1603-design-parking-system.swift
class ParkingSystem {
    var big: Int
    var medium: Int
    var small: Int

    init(_ big: Int, _ medium: Int, _ small: Int) {
        self.big = big
        self.medium = medium
        self.small = small
    }
    
    func addCar(_ carType: Int) -> Bool {
        if carType == 1 {
            if big == 0 {
                return false
            }
            big -= 1
            return true
        } else if carType == 2 {
            if medium == 0 {
                return false
            }
            medium -= 1
            return true
        } else {
            if small == 0 {
                return false
            }
            small -= 1
            return true
        }
    }
}

/**
 * Your ParkingSystem object will be instantiated and called as such:
 * let obj = ParkingSystem(big, medium, small)
 * let ret_1: Bool = obj.addCar(carType)
 */